### PR TITLE
Update Customer data with REST API

### LINKED ISSUE #21237: Update Customer data with REST API
### Preconditions (*)

1. 2.3.0 (have not tested previous versions)

### Steps to reproduce (*)

1. When using the API to update a customer

     curl -X PUT "https://example.com/index.php/rest/all/V1/customers/8" -H "accept: application/json" -H "Content-Type: application/json" -H "Authorization: Bearer TOKEN" -d "{ \"customer\": { \"middlename\": \"string\" }}"

### Expected result (*)

1. The customer should be updated for those fields just as when using the API to update a product  a PUT request like this 

    curl -X PUT "https://example.com/index.php/rest/all/V1/products/24-MB01" -H "accept: application/json" -H "Content-Type: application/json" -H "Authorization: Bearer TOKEN" -d "{ \"product\": { \"visibility\": 3 }}"

updates the visibility for the product with that SKU (as one would expect) however when i try to make a similar update to a customer with a PUT request i get:
### Actual result (*)

1. I get a 400 error with `"message": "The customer email is missing. Enter and try again."` the API to update only works if i add all of the required data like this:

    curl -X PUT "https://example.com/index.php/rest/all/V1/customers/8" -H "accept: application/json" -H "Content-Type: application/json" -H "Authorization: Bearer TOKEN" -d "{ \"customer\": { \"id\": \"8\", \"firstname\": \"Jon\", \"lastname\": \"Smith\", \"email\": \"Jjsmith@gmail.com\", \"middlename\": \"jac\", \"store_id\": 1, \"website_id\": 1 }}"

if I leave any of these values out i get an error with the exception of `id` which if left out creates a new record or throws an error if the email already exists (despite the `id` being specified at the end of the URL) also when `id` is passed in with the data that IDs record is updated even if it is not the `id` at the end of the URL

This does not seem to be the way the API should operate any help or explanation of whats going on would be appreciated.

### PR DESCRIPTION
**Pull request by engcom-Charlie**
### Description (*)

Fixed issue and tests

### Fixed Issues (if relevant)

1. Fixes magento/magento2#21237

### Manual testing scenarios (*)

See  magento/magento2#21237

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

File: app/code/Magento/Customer/Model/Plugin/UpdateCustomerId.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+     * @param string|null $passwordHash
+     * @return void
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function beforeSave(
+        CustomerRepositoryInterface $customerRepository,
+        CustomerInterface $customer,
+        ?string $passwordHash = null
+    ): void {
+        $cartId = $this->request->getParam('customerId');
```
Comment (Serializator) [inline app/code/Magento/Customer/Model/Plugin/UpdateCustomerId.php]: `$cartId` instead of `$customerId`

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
Hi @dmytro-ch, can You please review this PR?

**Review by ihor-sviziev — COMMENTED** [review]
✔ Approved.

**Review by ihor-sviziev — APPROVED** [review]
✔ Approved.

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-7618](https://jira.corp.magento.com/browse/ENGCOM-7618) has been created to process this Pull Request

**Review by dmytro-ch — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @dmytro-ch, thank you for the review.
[ENGCOM-7618](https://jira.corp.magento.com/browse/ENGCOM-7618) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Precondition**

Have one created customer

**Manual testing scenario:**

Send **PUT** request `/rest/V1/customers/{customer_id}` to update the customer

Screenshot from 2020-06-10 10-18-00


**Before:** :heavy_multiplication_x: 400. The customer email is missing

Screenshot from 2020-06-10 10-38-49

**After:** :heavy_check_mark: The customer was successfully updated

Screenshot from 2020-06-10 10-18-14

**Pull request merged**